File: packages/gbnf-experiment/data/20260909T144331Z_5b46f630/ported_implementation/src/utils/codePointLength.ts
// Python strings are sequences of code points, JavaScript strings are sequences of UTF-16
// code units. Positions reported by the parser are code point offsets, so anywhere the
// reference implementation uses `len(str)` for a position we count code points instead of
// using `String.prototype.length`.
export const codePointLength = (src: string): number => {
  let length = 0;
  for (const _ of src) {
    length += 1;
  }
  return length;
};

File: proposed-projects/round-trip-experiment/reverse/20260909T130706Z_31ac2789/20260910T073404Z_10c90cd6/ported_implementation/gbnf/utils/code_point_length.py
"""The number of code points in a string.

Python strings are sequences of code points, so this is simply ``len``. The
function exists so that positions, which are code point based throughout the
library, are always counted through a single, explicit helper.
"""


def code_point_length(src: str) -> int:
    return len(src)


codePointLength = code_point_length
